### PR TITLE
Print error SQL when throwing exception.SQLParsingException

### LINKED ISSUE #18455: Print error SQL when throwing exception.SQLParsingException
Hi community：
     Now，the ShardingSphere can print exception log when we want to excute error SQL.
     For example Cause: com.google.common.util.concurrent.UncheckedExecutionException: org.apache.shardingsphere.sql.parser.exception.SQLParsingException: You have an error in your SQL syntax.
    However, when throwing error sql exception, the shardingSphere cant print error sql needing to parser   to help developer modify sql. I hope the shardingSphere can have the feature. Now,"sql-show:true "cant achieve the goal.  
    My shardingSphere version is 5.0.0

### PR DESCRIPTION
**Pull request by ninjaboy97**
Fixes #18455 .

Changes proposed in this pull request:
- Throw root cause of SQLParsingException

### CODE REVIEW

File: shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java [hunk 1]
```diff
@@ -40,12 +40,14 @@ public SQLStatement parse(final String sql) {
...
+
     private ASTNode parseToASTNode(final String sql) {
         try {
             return SQLParserFactory.newInstance(sql, AdvancedDistSQLLexer.class, AdvancedDistSQLParser.class).parse();
-        } catch (final ParseCancellationException | SQLParsingException ignored) {
-            throw new SQLParsingException("You have an error in your SQL syntax.");
+        } catch (final ParseCancellationException ignored) {
+            throw new SQLParsingException("You have an error in your SQL syntax: " + sql);
+        } catch (final SQLParsingException sqlParsingException) {
+            throw sqlParsingException;
```
Comment (terrymanu) [inline shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java:50]: Why separate this two exception, how about just throw together?
Does the ParseCancellationException include SQL and parse error cause?

Comment (ninjaboy97) [reply shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java:50]: The ParseCancellationException does not include SQL and parse error cause.

Comment (ninjaboy97) [inline shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java:50]: I couldn't test this line from the tests. Any invalid sql I generate throws ParseCancellationException and therefore line 48 is being tested. Any advice on how I can test this line @terrymanu?

Comment (terrymanu) [reply shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java:50]: ParseCancellationException maybe throw from parse mode error

Comment (terrymanu) [reply shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java:50]: Can you try to input an error SQL from JDBC and Proxy, and parse the exception here.
I want to make sure the exception thrown on the right way.

**Comment by terrymanu** [comment]
I'am expecting the screenshot for exception thrown when input type SQL

**Comment by ninjaboy97** [comment]
Hey! I'll be able to update this by the coming weekend.

**Comment by terrymanu** [comment]
@ninjaboy97 Hi, any update?

**Comment by terrymanu** [comment]
@ninjaboy97 The progress of this PR has not been upgraded for a long time. If there is no update, I will close it soon.

**Comment by terrymanu** [comment]
I just close this PR and reopen issue #18455.

**Pull request closed without merging**